internal/repometadata/repometadata_test.go

```diff
@@ -0,0 +1,318 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repometadata
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/googleapis/librarian/internal/config"
+)
+
+func TestGenerateRepoMetadata(t *testing.T) {
+	// Create a test service YAML
+	yamlContent := `type: google.api.Service
+config_version: 3
+name: secretmanager.googleapis.com
+title: Secret Manager API
+
+documentation:
+  summary: |-
+    Stores sensitive data such as API keys, passwords, and certificates.
+    Provides convenience while improving security.
+
+publishing:
+  documentation_uri: https://cloud.google.com/secret-manager/docs/overview
+  api_short_name: secretmanager
+`
+
+	tmpDir := t.TempDir()
+	serviceYAMLPath := filepath.Join(tmpDir, "service.yaml")
+	if err := os.WriteFile(serviceYAMLPath, []byte(yamlContent), 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	outDir := filepath.Join(tmpDir, "output")
+	if err := os.MkdirAll(outDir, 0755); err != nil {
+		t.Fatal(err)
+	}
+
+	library := &config.Library{
+		Name:         "google-cloud-secret-manager",
+		ReleaseLevel: "stable",
+	}
+
+	if err := GenerateRepoMetadata(library, "python", "googleapis/google-cloud-python", serviceYAMLPath, outDir, []string{"google/cloud/secretmanager/v1", "google/cloud/secretmanager/v2"}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Read back the generated metadata
+	data, err := os.ReadFile(filepath.Join(outDir, ".repo-metadata.json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var got RepoMetadata
+	if err := json.Unmarshal(data, &got); err != nil {
+		t.Fatal(err)
+	}
+
+	want := RepoMetadata{
+		Name:                 "secretmanager",
+		NamePretty:           "Secret Manager",
+		ProductDocumentation: "https://cloud.google.com/secret-manager/",
+		ClientDocumentation:  "https://cloud.google.com/python/docs/reference/secretmanager/latest",
+		IssueTracker:         "",
+		ReleaseLevel:         "stable",
+		Language:             "python",
+		LibraryType:          "GAPIC_AUTO",
+		Repo:                 "googleapis/google-cloud-python",
+		DistributionName:     "google-cloud-secret-manager",
+		APIID:                "secretmanager.googleapis.com",
+		DefaultVersion:       "v2",
+		APIShortname:         "secretmanager",
+		APIDescription:       "Stores sensitive data such as API keys, passwords, and certificates.\nProvides convenience while improving security.",
+	}
+
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("mismatch (-want +got):\n%s", diff)
+	}
+}
+
+func TestGenerateRepoMetadata_WithOverrides(t *testing.T) {
```

Comment:
I took this from the librarianx repo, where there are two separate test functions - but I can refactor it to a table-driven test if you'd prefer.